compiler/scalc/to_c.ml

```diff
@@ -0,0 +1,242 @@
+(* This file is part of the Catala compiler, a specification language for tax
+   and social benefits computation rules. Copyright (C) 2020 Inria, contributor:
+   Denis Merigoux <denis.merigoux@inria.fr>
+
+   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+   use this file except in compliance with the License. You may obtain a copy of
+   the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+   License for the specific language governing permissions and limitations under
+   the License. *)
+
+open Catala_utils
+open Shared_ast
+module Runtime = Runtime_ocaml.Runtime
+module D = Dcalc.Ast
+module L = Lcalc.Ast
+
+let avoid_keywords (s : string) : string =
+  if
+    match s with
+    (* list taken from
+       https://learn.microsoft.com/en-us/cpp/c-language/c-keywords *)
+    | "auto" | "break" | "case" | "char" | "const" | "continue" | "default"
+    | "do" | "double" | "else" | "enum" | "extern" | "float" | "for" | "goto"
+    | "if" | "inline" | "int" | "long" | "register" | "restrict" | "return"
+    | "short" | "signed" | "sizeof" | "static" | "struct" | "switch" | "typedef"
+    | "union" | "unsigned" | "void" | "volatile" | "while" ->
+      true
+    | _ -> false
+  then s ^ "_"
+  else s
+
+let format_struct_name (fmt : Format.formatter) (v : StructName.t) : unit =
+  Format.fprintf fmt "%s"
+    (Format.asprintf "%a" StructName.format v
+    |> String.to_ascii
+    |> String.to_snake_case
+    |> avoid_keywords)
+
+let format_struct_field_name (fmt : Format.formatter) (v : StructField.t) : unit
+    =
+  Format.fprintf fmt "%s"
+    (Format.asprintf "%a" StructField.format v
+    |> String.to_ascii
+    |> String.to_snake_case
+    |> avoid_keywords)
+
+let format_enum_name (fmt : Format.formatter) (v : EnumName.t) : unit =
+  Format.fprintf fmt "%s"
+    (Format.asprintf "%a" EnumName.format v
+    |> String.to_ascii
+    |> String.to_snake_case
+    |> avoid_keywords)
+
+let format_enum_cons_name (fmt : Format.formatter) (v : EnumConstructor.t) :
+    unit =
+  Format.fprintf fmt "%s"
+    (Format.asprintf "%a" EnumConstructor.format v
+    |> String.to_ascii
+    |> String.to_snake_case
+    |> avoid_keywords)
+
+let rec format_typ (fmt : Format.formatter) (typ : typ) : unit =
+  match Mark.remove typ with
+  | TLit TUnit -> Format.fprintf fmt "void * /* Unit */"
+  | TLit TMoney -> Format.fprintf fmt "int"
+  | TLit TInt -> Format.fprintf fmt "int"
+  | TLit TRat -> Format.fprintf fmt "double"
+  | TLit TDate -> Format.fprintf fmt "double"
+  | TLit TDuration -> Format.fprintf fmt "double"
```

Comment:
Shouldn't we use gmp's bigint and rational numbers here ?

Reply:
Yes of course we should but I'll deal with those later, first I want to have a C syntactic skeleton running